client/signup/main.jsx

```diff
@@ -298,11 +301,23 @@ class Signup extends React.Component {
 		const flowSteps = flows.getFlow( flowName ).steps;
 		map( flowSteps, flowStepName => this.processFulfilledSteps( flowStepName, nextProps ) );
 
-		if ( includes( flows.excludedSteps, stepName ) ) {
+		if ( flows.excludedSteps.has( stepName ) ) {
 			this.goToNextStep( flowName );
 		}
 	};
 
+	addUnfulfilledSteps( nextProps ) {
+		for ( const stepName of flows.excludedSteps ) {
+			if ( ! flows.getFlows()[ nextProps.flowName ].steps.includes( stepName ) ) {
```

Comment:
This condition is a lot to digest at a glance.
Could we split this up in to more readable chunks, perhaps something like:
```suggestion
			const flow = flows.getFlows()[ nextProps.flowName ];
			if ( flow && flow.steps.includes( stepName ) ) {
```

That's also a little safer to execute, though if the flow is missing at this point we have other issues!

Reply:
Fair, I've broken that line up